### PR TITLE
Extract some reused SCALE en/decoding logic into `utils`

### PR DESCRIPTION
**Pull request by David-Petrov**
A beginning of a vision to extract away all the SCALE encoding/decoding logic into `utils`.

### CODE REVIEW

**Review by bokoto000 — APPROVED** [review]
lgtm

File: src/main/java/com/limechain/utils/scale/ScaleUtils.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+@UtilityClass
+public class ScaleUtils {
+
+    @UtilityClass
+    public class Decode {
+        public <T> T decode(byte[] encodedData, ScaleReader<T> reader) {
+            try {
+                return new ScaleCodecReader(encodedData).read(reader);
+            } catch (RuntimeException e) {
+                throw new RuntimeException("Error while SCALE decoding.", e); // TODO: is this a code smell?
```
Comment (georg-getz) [inline src/main/java/com/limechain/utils/scale/ScaleUtils.java]: Yes, we'll talk about this in tomorrow's daily but we won't throwing `RuntimeException` anymore. Extend it with a custom more specific one and throw that one instead.

File: src/main/java/com/limechain/utils/scale/ScaleUtils.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Function;
+
+// TODO: This class is currently just a helper utility class,
+//  planned to grow into a unified scale encode/decode util class with whatever methods are useful
+//  WIP
+// Currently trying out different approaches to spare some of the boilerplate around SCALE en/decoding
```
Comment (georg-getz) [inline src/main/java/com/limechain/utils/scale/ScaleUtils.java]: Indentation is a big weird, why not have it all start at the left. You don't need a comma after class and you delete the first "class" and the "just" strings.

**Review by vikinatora — APPROVED** [review]

File: src/main/java/com/limechain/utils/scale/ScaleUtils.java (hunk 1 above)
Comment (David-Petrov) [reply src/main/java/com/limechain/utils/scale/ScaleUtils.java]: Done. 

Also, while writing the custom ScaleDecodingException, I noticed there was an existing ScaleEncodingException in `network.protocol.warp.exception`, moved it into `utils.scale.exceptions`.
If nobody has anything against this, we can resolve this conversation and merge the PR.

**Review by georg-getz — APPROVED** [review]

**Pull request merged**